Formula/sqlcmd.rb

```diff
@@ -0,0 +1,21 @@
+class Sqlcmd < Formula
+  desc "Microsoft SQL Server command-line interface"
+  homepage "https://github.com/microsoft/go-sqlcmd"
+  url "https://github.com/microsoft/go-sqlcmd.git",
+  tag:      "v0.8.1",
+  revision: "4e0b95ce49b8164c6496ca1b7a85fa57734fef4c"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["CGO_ENABLED"] = "0"
+
+    system "go", "build", *std_go_args(ldflags: "-s -w"), "./cmd/sqlcmd"
+  end
+
+  test do
+    out = shell_output("#{bin}/sqlcmd -?")
+    assert_match "Usage: sqlcmd", out
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Pushed change to follow the docker.rb approach.